flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/agg/AggsHandlerCodeGenerator.scala

```diff
@@ -1275,8 +1277,15 @@ class AggsHandlerCodeGenerator(
       needReset: Boolean = false,
       needEmitValue: Boolean = false): Unit = {
     // check and validate the needed methods
-    aggBufferCodeGens.foreach(
-      _.checkNeededMethods(needAccumulate, needRetract, needMerge, needReset, needEmitValue))
+    aggBufferCodeGens.zipWithIndex.foreach {
```

Comment:
This is comment on function `genRetract`: IIUC, `checkNeededMethods` only check whether there is a `retract` function in UDX, but not affect generating code with `retract`. Should we also update the logic about codegen by `aggActionCodeGens`？

Reply:
how to change the logic of codegen by aggActionCodeGens? i don't understand